src/python/CRABClient/Commands/checkfile.py

```diff
@@ -0,0 +1,268 @@
+        if ec:
+            if 'File not found' in err or 'No such file' in err:
+                msg = "File not found"
+                return False, msg
+        try:
+            replicaSize = int(out.split()[4])
+        except Exception:  # pylint: disable=broad-except
+            msg = "Error looking up file at remote site"
+            return False, msg
+        if replicaSize != self.fileToCheck['size']:
+            msg = f"Replica size at remote side is wrong. {replicaSize} vs. {self.fileToCheck['size']}"
+            msg += "  Needs to be invalidated"
+            return False, msg
+        return True, ""
+
+    def checkReplicaAdler32(self, pfn):
+        """
+        compute Adler32 on a remote PFN
+        """
+
+        f = tempfile.NamedTemporaryFile(delete=False, prefix='testFile_')
+        fname = f.name
+        execute_command(f"rm -f {fname}")
+        print("copy file...", end="", flush=True)
+        cmd = f"eval `scram unsetenv -sh`; gfal-copy {pfn} {fname}"
+        out, err, ec = execute_command(cmd)
+        if ec:
+            msg = f"file copy for checksum control failed.\n{out}\n{err}"
+            execute_command(f"rm -f {fname}")
+            return False, msg
+        print("compute Adler32 checksum...", end="", flush=True)
+        cmd = f"eval `scram unsetenv -sh`; gfal-sum {fname} ADLER32"
+        out, err, ec = execute_command(cmd)
+        if ec:
+            msg = f"checksum computation failed.\n{out}\n{err}"
+            execute_command(f"rm -f {fname}")
+            return False, msg
+        print("Done")
+        adler32 = out.split()[1]
+        if not adler32 == self.fileToCheck['adler32']:
+            msg = f"Remote replica has wrong checksum. {adler32} vs. {self.fileToCheck['adler32']}"
+            msg += "  Needs to be invalidated"
+            execute_command(f"rm -f {fname}")
+            return False, msg
+        execute_command(f"rm -f {fname}")
+        return True, ""
+
+    def setOptions(self):
+        """
+        __setOptions__
+
+        This allows to set specific command options
+        """
+        self.parser.add_option('--lfn',
+                               dest='lfn',
+                               default=None,
+                               help='LFN of the file to check')
+        self.parser.add_option('--checksum',
+                               dest='checkChecksum',
+                               action="store_true",
+                               help='check checksum of all disk replicas. SLOW !')
+        self.parser.add_option('--instance', dest='instance', default='prod/global',
+                               help="DBS instance. e.g. prod/global (default) or prod/phys03"
+                               )
+        self.parser.add_option('--scope', dest='scope', default=None,
+                               help="Rucio scope. Default is 'cms' for global DBS and 'user:username' for Phys03"
+                               )
```

Comment:
imagine you have your own dataset described in phys03.. need to get DBS info from there and rucio scope from user.<username>. But so far those are more a reminder of looking a this. As I said I did not test, nor really thought well the use case.

Reply:
OK. The thing is that if you want to check a USER file in PHYS03, you do not only need to indicate this DBS instance, but since the file may belong to another user, you also need to indicate the Rucio scope. We could extract the scope from the file name, but I am not sure that a DID can not be moved or replicated from one scope to another. So I left it to the user to indicate it in this case. Anyhow at this moment those are "expert options"

Reply:
Ah...sorry. I just want to suggest to rename option from `--instance` to `--dbs-instance`.

Reply:
I see  ! Thanks. Good idea. Indeed elsewhere we use --instance to indicate crab prod/preprod/.... but I should also change in crab setdatasetstatus and setfilestatus ! I will review all uses